examples/vision/depth_estimation.py

```diff
@@ -0,0 +1,507 @@
+"""
+Title: An approach towards Depth Estimation with CNN
+Author: [Victor Basu](https://www.linkedin.com/in/victor-basu-520958147)
+Date created: 2021/08/09
+Last modified: 2021/08/10
+Description: Implement an depth estimation model with CNN.
+"""
+"""
+Title: An approach towards Depth Estimation with CNN.
+
+Author: [Victor Basu](https://www.linkedin.com/in/victor-basu-520958147)
+
+Date created: 2021/08/08
+
+Last modified: 2021/08/09
+
+Description: Implement an depth estimation model with CNN.
+"""
+
+"""
+## Introduction
+
+**Depth Estimation** is a crucial step towards inferring scene geometry from 2D images.
+The goal in monocular Depth Estimation is to predict the depth value of each pixel, given
+only a single RGB image as input.
+
+This is an approach to build a depth estimation model with CNN and basic loss functions.
+While I was working on this topic I came across various research paper that explains
+solving this problem and to be honest my solution is not as good as in those papers. I
+would suggest you to go through those, just search depth estimation on
+"paperwithcode.com".
+
+![depth](https://paperswithcode.com/media/thumbnails/task/task-0000000605-d9849a91.jpg)
+
+
+"""
+
+"""
+##Setup
+"""
+
+import os
+import cv2
+import csv
+import logging
+import sys
+
+import tensorflow as tf
+from tensorflow.keras import layers
+import tensorflow.keras.backend as K
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from PIL import Image
+from sklearn.utils import shuffle
```

Comment:
you can do shuffling without sklearn.shuffle.

Reply:
updated